tweetflask.py

```diff
@@ -24,41 +29,66 @@ def get():
+    return jsonify(get_tweet(jsondata, None, None, None))
+
+
+@app.route('/tweets/<get_id>')
+def get_ID(get_id):
+    # Opening and reading the db file
+    jsondata = read_file()
+
+    for tweets in jsondata['TweetData']:
+        if get_id == tweets['TweetID']:
+            for user in jsondata['UserData']:
+                if tweets['UserID'] == user['UserID']:
+                    return jsonify(get_tweet(jsondata, user, None, get_id))
+    return jsonify(errors('404', 'tweet does not exist')), 404
 
 
 # looks up all the tweets that belong to the username provided
 # and looks for the mood if provided
-def get_tweet(jsondata, user, mood):
+def get_tweet(jsondata, user, mood, get_id):
     tweet_data = {
         'data': [
         ]
     }
+
     if user is None:
         return tweet_data
     for tweets in jsondata['TweetData']:
         if user['UserID'] == tweets['UserID']:
-            if not mood or mood.lower() == tweets['Mood'].lower():
-                response_data = {}
-                response_data['id'] = tweets['TweetID']
-                response_data['type'] = 'tweet'
-                time = datetime.strptime(tweets['TimeAndDate'],
-                                         '%d-%b-%y %I.%M.%S.%f %p %z')
-                response_data['attributes'] = {
-                    'username': user['Username'],
-                    'userFullName': user['UserFullName'],
-                    'timeAndDate': time.isoformat(),
-                    'title': tweets['Title'],
-                    'mood': tweets['Mood'],
-                    'tweet': tweets['TweetMsg']
-                }
-                tweet_data['data'].append(response_data)
+            if not get_id or get_id == tweets['TweetID']:
+                if not mood or mood.lower() == tweets['Mood'].lower():
+                    response_data = {}
+                    response_data['id'] = tweets['TweetID']
+                    response_data['type'] = 'tweet'
+                    time = datetime.strptime(tweets['TimeAndDate'],
+                                             '%d-%b-%y %I.%M.%S.%f %p %z')
+                    tweet_data['data'].append(attributes(response_data,
+                                                         user['Username'],
+                                                         user['UserFullName'],
+                                                         time,
+                                                         tweets['Title'],
+                                                         tweets['Mood'],
+                                                         tweets['TweetMsg']))
     return tweet_data
 
 
+def attributes(response_data, username, userFullName, time, title, mood, msg):
+    response_data['attributes'] = {
+        'username': username,
+        'userFullName': userFullName,
+        'timeAndDate': time.isoformat(),
+        'title': title,
+        'mood': mood,
+        'tweet': msg
+    }
+    return response_data
```

Comment:
I don't think you need to feed an empty dictionary `response_data` as an input to this function. Also, some parameters can be combined as one, like `user['Username']` and `user['UserFullName']`; `tweets['Title']`, `tweets['Mood']` and `tweets['TweetMsg']`.